tests/unit/test_log_proxy_consumer.py

```diff
@@ -167,6 +167,12 @@ def test__add_client(self):
             {x["url"] for x in self.harness.charm._log_proxy._clients_list()}, expected_clients
         )
 
+    def test__empty_clients_list(self):
+        # Ensure we do not raise an exception if the clients list is empty
+        rel_id = self.harness.add_relation("log-proxy", "agent")
+        self.harness.add_relation_unit(rel_id, "agent/0")
+        self.assertEqual(self.harness.charm._log_proxy._clients_list(), [])
```

Comment:
Is there a more indirect way of testing this? Prefer blackbox testing over whitebox testing.
For example:
```
class TestRelatedEndpoints:
    def test_no_endpoints
    def test_one_invalid_endpoint
    def test_endpoints_without_scheme
```

Reply:
I agree with Leon on this. Can we assess this using a public API instead?